chapters/arrays.tex

```diff
@@ -803,25 +803,17 @@ \subsubsection{Constructor with Several Iterators}\label{array-constructor-with-
 
 \subsection{Concatenation}\label{array-concatenation}\label{concatenation}
 
-The function \lstinline!cat($k$, A, B, C, $\ldots$)! concatenates arrays
-\lstinline!A!, \lstinline!B!, \lstinline!C!, \ldots along
-dimension $k$ according to the following rules:
+The function \lstinline!cat($k$, A, B, C, $\ldots$)!\indexinline{cat} concatenates arrays \lstinline!A!, \lstinline!B!, \lstinline!C!, \ldots along dimension $k$ according to the following rules:
 \begin{itemize}
 \item
-  Arrays \lstinline!A!, \lstinline!B!, \lstinline!C!, \ldots must have the same number of dimensions, i.e.,
-  \lstinline!ndims(A)! = \lstinline!ndims(B)! = \ldots
+  Arrays \lstinline!A!, \lstinline!B!, \lstinline!C!, \ldots must have the same number of dimensions, i.e., \lstinline!ndims(A)! = \lstinline!ndims(B)! = \ldots
 \item
-  Arrays \lstinline!A!, \lstinline!B!, \lstinline!C!, \ldots must be type compatible expressions (\cref{type-compatible-expressions})
-  giving the type of the elements of the result. The maximally expanded
-  types should be equivalent. \lstinline!Real! and \lstinline!Integer! subtypes can be mixed
-  resulting in a \lstinline!Real! result array where the \lstinline!Integer! numbers have been
-  transformed to \lstinline!Real! numbers.
+  Arrays \lstinline!A!, \lstinline!B!, \lstinline!C!, \ldots must be type compatible expressions (\cref{type-compatible-expressions}) giving the type of the elements of the result. The maximally expanded types should be equivalent.
```

Comment:
```suggestion
  Arrays \lstinline!A!, \lstinline!B!, \lstinline!C!, \ldots must be type compatible expressions (\cref{type-compatible-expressions}) giving the type of the elements of the result.
The maximally expanded types should be equivalent.
```
If we are going for sentence-based lines we should do it consistently.

Reply:
Of course, thanks!